src/enc/brotli_bit_stream.rs

```diff
@@ -2113,49 +2115,52 @@ fn Context(p1: u8, p2: u8, mode: ContextType) -> u8 {
-    xself: &mut BlockEncoder<Alloc>,
-    symbol: usize,
-    context: usize,
-    context_map: &[u32],
-    storage_ix: &mut usize,
-    storage: &mut [u8],
-    context_bits: usize,
-) {
-    if xself.block_len_ == 0usize {
-        let block_ix: usize = {
-            xself.block_ix_ = xself.block_ix_.wrapping_add(1);
-            xself.block_ix_
-        };
-        let block_len: u32 = xself.block_lengths_[block_ix];
-        let block_type: u8 = xself.block_types_[block_ix];
-        xself.block_len_ = block_len as usize;
-        xself.entropy_ix_ = (block_type as usize) << context_bits;
-        StoreBlockSwitch(
-            &mut xself.block_split_code_,
-            block_len,
-            block_type,
-            0i32,
-            storage_ix,
-            storage,
-        );
-    }
-    xself.block_len_ = xself.block_len_.wrapping_sub(1);
-    {
-        let histo_ix: usize = context_map[xself.entropy_ix_.wrapping_add(context)] as usize;
-        let ix: usize = histo_ix
-            .wrapping_mul(xself.histogram_length_)
-            .wrapping_add(symbol);
-        BrotliWriteBits(
-            xself.depths_.slice()[ix],
-            xself.bits_.slice()[ix] as (u64),
-            storage_ix,
-            storage,
-        );
+impl<Alloc: Allocator<u8> + Allocator<u16>> BlockEncoder<'_, Alloc> {
+    fn store_symbol_with_context(
+        &mut self,
+        symbol: usize,
+        context: usize,
+        context_map: &[u32],
+        storage_ix: &mut usize,
+        storage: &mut [u8],
+        context_bits: usize,
+    ) {
+        if self.block_len_ == 0 {
+            let block_ix: usize = {
+                self.block_ix_ = self.block_ix_.wrapping_add(1);
+                self.block_ix_
+            };
+            let block_len: u32 = self.block_lengths_[block_ix];
+            let block_type: u8 = self.block_types_[block_ix];
+            self.block_len_ = block_len as usize;
+            self.entropy_ix_ = (block_type as usize) << context_bits;
+            StoreBlockSwitch(
+                &mut self.block_split_code_,
+                block_len,
+                block_type,
+                0,
+                storage_ix,
+                storage,
+            );
+        }
+        self.block_len_ = self.block_len_.wrapping_sub(1);
+        {
+            let histo_ix: usize = context_map[self.entropy_ix_.wrapping_add(context)] as usize;
+            let ix: usize = histo_ix
+                .wrapping_mul(self.histogram_length_)
+                .wrapping_add(symbol);
+            BrotliWriteBits(
+                self.depths_.slice()[ix],
+                self.bits_.slice()[ix] as (u64),
+                storage_ix,
+                storage,
+            );
+        }
     }
 }
 
 fn CommandCopyLen(xself: &Command) -> u32 {
+    // BUG? There are 3 implementations of this method with different constant value
     xself.copy_len_ & 0xffffffu32
```

Comment:
Hmm I changed this in 29d5ad625e4cae79d6050365c62edcf27321dc11 in supporting larger window sizes...

Reply:
thx, i merged all of them now that they are all the same - see https://github.com/dropbox/rust-brotli/pull/168